test/unit/DetectChanges/BCBreak/ClassBased/PropertyChangedTest.php

```diff
@@ -80,4 +80,91 @@ class TheClass {
             )
         );
     }
+
+    /**
+     * @dataProvider finalClassProvider
+     */
+    public function testWillIgnoreChangeInFinalClass($fromLocator, $toLocator) : void
+    {
+
+
+        $comparator = $this->createMock(PropertyBased::class);
+
+        $comparator
+            ->expects(self::exactly(2))
+            ->method('__invoke')
+            ->willReturnCallback(static function (ReflectionProperty $from, ReflectionProperty $to) : Changes {
+                $propertyName = $from->getName();
+
+                self::assertSame($propertyName, $to->getName());
+
+                return Changes::fromList(Change::added($propertyName, true));
+            });
+
+        Assertion::assertChangesEqual(
+            Changes::empty(),
+            (new PropertyChanged($comparator))->__invoke(
+                (new ClassReflector($fromLocator))->reflect('TheClass'),
+                (new ClassReflector($toLocator))->reflect('TheClass')
+            )
+        );
+    }
+
+    public function finalClassProvider() : array
+    {
+        $astLocator = (new BetterReflection())->astLocator();
+
+        return [
+            'final class' => [
+                new StringSourceLocator(
+                    <<<'PHP'
+<?php
+
+final class TheClass {
+    protected $b;
+}
+PHP
+                    ,
+                    $astLocator
+                ),
+                new StringSourceLocator(
+                    <<<'PHP'
+<?php
+final class TheClass {
+    private $b;
```

Comment:
In this case, I'd say that the property was indeed changed: this shouldn't produce a `Changes::empty()`

Reply:
Why? it doesn't make sense to have a protected property in this case. nobody can access it only the current class, so this is not a bc break. since private and protected are equal in this case. Since there is no parent class.

Reply:
`PropertyChanged` forwards checks to futher BCBreak checkers: it doesn't really raise a BC break itself.

In the config of this project, changes to `final` classes are scoped within the `FinalClassChanged` checks:

https://github.com/Roave/BackwardCompatibilityCheck/blob/3dd77835db29c46e571c41acedf733766015b871/bin/roave-backward-compatibility-check.php#L182

I think what's being hit is this other checker: 

https://github.com/Roave/BackwardCompatibilityCheck/blob/3dd77835db29c46e571c41acedf733766015b871/bin/roave-backward-compatibility-check.php#L83-L90

As you can see, not scoped within `final` classes context: we probably want such a check to only apply to open classes, or to `public` properties of `final` classes, so the rule has to be nested a bit deeper.